src/kpro_compress.erl

```diff
@@ -0,0 +1,87 @@
+%%%   Copyright (c) 2018, Klarna AB
+%%%
+%%%   Licensed under the Apache License, Version 2.0 (the "License");
+%%%   you may not use this file except in compliance with the License.
+%%%   You may obtain a copy of the License at
+%%%
+%%%       http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%%   Unless required by applicable law or agreed to in writing, software
+%%%   distributed under the License is distributed on an "AS IS" BASIS,
+%%%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%%   See the License for the specific language governing permissions and
+%%%   limitations under the License.
+%%%
+
+-module(kpro_compress).
+
+-export([ compress/2
+        , decompress/2
+        , codec_to_method/1
+        , method_to_codec/1
+        ]).
+
+-include("kpro_private.hrl").
+
+%% @doc Translate codec in kafka batch attributes to compression method.
+-spec codec_to_method(byte()) -> kpro:compress_option().
+codec_to_method(A) when ?KPRO_IS_GZIP_ATTR(A) -> ?gzip;
+codec_to_method(A) when ?KPRO_IS_SNAPPY_ATTR(A) -> ?snappy;
+codec_to_method(A) when ?KPRO_IS_LZ4_ATTR(A) -> ?lz4;
+codec_to_method(_) -> ?no_compression.
+
+%% @doc Translate compression method to bits for kafka batch attributes.
+method_to_codec(?gzip) -> ?KPRO_COMPRESS_GZIP;
+method_to_codec(?snappy) -> ?KPRO_COMPRESS_SNAPPY;
+method_to_codec(?lz4) -> ?KPRO_COMPRESS_LZ4;
+method_to_codec(?no_compression) -> ?KPRO_COMPRESS_NONE.
+
+%% @doc Compress encoded batch.
+-spec compress(kpro:compress_option(), iodata()) -> iodata().
+compress(?no_compression, IoData) -> IoData;
+compress(?gzip, IoData)           -> zlib:gzip(IoData);
+compress(?snappy, IoData)         -> snappy_compress(IoData).
```

Comment:
omitted deliberately.
supported the name, but compression/decompression are not implemented
for more clean crash log.
otherwise crashing on decoding the 'lz4' name would be confusing.